desci-server/src/controllers/doi/check.ts

```diff
@@ -0,0 +1,104 @@
+import { DocumentId } from '@automerge/automerge-repo';
+import { CommonComponentPayload, PdfComponent, ResearchObjectComponentType } from '@desci-labs/desci-models';
+import { NextFunction, Request, Response } from 'express';
+
+import { DoiError } from '../../core/doi/error.js';
+import {
+  BadRequestError,
+  ForbiddenError,
+  NotFoundError,
+  RequestWithNode,
+  SuccessResponse,
+  crossRefClient,
+  doiService,
+  getLatestManifestFromNode,
+  logger as parentLogger,
+} from '../../internal.js';
+import { WorkSelectOptions } from '../../services/crossRef/definitions.js';
+import repoService from '../../services/repoService.js';
+
+export const checkMintability = async (req: Request, res: Response, _next: NextFunction) => {
+  const { uuid } = req.params;
+  if (!uuid) throw new BadRequestError();
+
+  const logger = parentLogger.child({
+    module: 'DOI::checkMintability',
+  });
+
+  try {
+    await doiService.checkMintability(uuid);
+    new SuccessResponse(true).send(res);
+  } catch (err) {
+    logger.error(err, 'module:: checkMintability');
+    if (!(err instanceof DoiError)) {
+      // TODO: Sentry error reporting
+    }
+    new SuccessResponse(false).send(res);
+  }
+};
+
+export const getDoi = async (req: Request, res: Response, next: NextFunction) => {
+  const { identifier } = req.params;
+  if (!identifier) throw new BadRequestError();
+
+  const doi = await doiService.getDoiByDpidOrUuid(identifier);
+  new SuccessResponse(doi).send(res);
+};
+
+export const attachDoi = async (req: RequestWithNode, res: Response, _next: NextFunction) => {
+  const { uuid, path } = req.body;
+
+  const logger = parentLogger.child({
+    module: 'DOI::AttachDOI',
+  });
+
+  const node = req.node;
+
+  const latestManifest = await getLatestManifestFromNode(node);
+  const componentIndex = latestManifest.components.findIndex(
+    (component) =>
+      component.type === ResearchObjectComponentType.PDF && (component.payload as CommonComponentPayload).path === path,
+  );
+
+  if (componentIndex === -1) throw new BadRequestError('Component to attach DOI not a valid pdf');
+
+  const component = latestManifest.components[componentIndex] as PdfComponent;
+
+  if (component.payload.doi) throw new ForbiddenError(`${component.subtype || component.type} already has a DOI`);
```

Comment:
Make sure DOI is an array, there can be multiple DOIs for a pdf

Reply:
I didn't catch this, I'll push the updates to a new branch